drivers/mtd/mtd_config.c

```diff
@@ -1807,7 +1807,7 @@ int mtdconfig_unregister(void)
 
   inode = file.f_inode;
   dev = (FAR struct mtdconfig_struct_s *)inode->i_private;
-  nxmutex_destroy(&dev->exclsem);
+  nxsem_destroy(&dev->exclsem);
```

Comment:
Since exclsem is used as resource count(ensure only one caller can open it at the same time):

1. Call nxsem_wait in mtdconfig_open
2. Call nxsem_post in mtdconfig_close

Only nxsem_wait/nxsem_post is called in pair can be replaced by mutex_lock/mutex_unlock.